### PR TITLE
Switchport Mode & CLI Modified Fix

### PR DESCRIPTION
**Pull request by sabakram**
#### What I did
This PR is Fixture for https://github.com/sonic-net/sonic-utilities/pull/3108 The PR was reverted due to backward compatibility issues. As per new suggestions, removed db migrator changes from this new change along with vlan.py & switchport.py changes

To Fix issues as per suggestions removed default mode from YANG model and removed minigraph changes:

#### How I did it
       1. Removed Db migrator changes from code. 
       2. Modified Vlan.py & Switchport.py changes 

#### How to verify it
New commands have been added in Command-Reference.md  All the syntax and examples have been added there and they can be verified by running the specific command

### CODE REVIEW

**Comment by sabakram** [comment]
Hi @wen587, can you please review this PR. We have modified code. We removed Db-migrator changes along with updated vlan.py & switchport.py for default mode behavior.

File: config/main.py
Comment (wen587) [inline config/main.py:4617]: else should be used

File: tests/vlan_test.py
Comment (wen587) [inline tests/vlan_test.py:1158]: Why the exit_code is different with that without change? It seems introduce backward compatibility issue.

**Review by wen587 — APPROVED** [review]
lgtm. Please check with other reviewer.

**Comment by ridahanif96** [comment]
> Hi @theasianpianist , @prsunny @venkatmahalingam can you please help review this PR. 

Hi All, 

Can you help with review this PR. Thanks.

**Review by theasianpianist — APPROVED** [review]
Vlan changes lgtm

**Comment by ridahanif96** [comment]
Hi @prsunny & @venkatmahalingam , can you please help with review. Thanks in advance,

**Comment by prsunny** [comment]
Can you please confirm there is no changes to existing 'show' commands? I see there is a new show CLI command which is ok.

**Comment by prsunny** [comment]
Please confirm, this issue is fixed in this PR - https://github.com/sonic-net/sonic-buildimage/issues/18392

**Comment by sabakram** [comment]
> Can you please confirm there is no changes to existing 'show' commands? I see there is a new show CLI command which is ok.

Hi @prsunny , we didn't make any change in exisiting show commands i.e in "show vlan brief" & "show int status". We proposed new show commands for switchport mode i.e "show int switchport config" & "show int switchport status"

**Comment by sabakram** [comment]
> Please confirm, this issue is fixed in this PR - [sonic-net/sonic-buildimage#18392](https://github.com/sonic-net/sonic-buildimage/issues/18392)

Yes, We have checked this change with latest build. Please see behavior 
ShowVersion
Cffgen_2
Cffgen_l2
Portchannel

**Comment by sabakram** [comment]
Hi @wen587 Do you also suggest change for this PR in [sonic_mgmt/gcu/test_vlan.py](https://github.com/sonic-net/sonic-mgmt/blob/master/tests/generic_config_updater/test_vlan_interface.py)?

**Review by prsunny — APPROVED** [review]

**Comment by ridahanif96** [comment]
@venkatmahalingam Hi Venkat, can you pls help review this PR.

**Review by ridahanif96 — APPROVED** [review]

File: utilities_common/cli.py
Comment (qiluo-msft) [inline utilities_common/cli.py:275]: >vid_range [](http://example.com/codeflow?start=26&length=9)

will it throw if there is no "-"  in vid_range?

File: utilities_common/cli.py
Comment (qiluo-msft) [inline utilities_common/cli.py:286]: >ctx.fail("Invalid VLAN ID must be in (2-4094)") [](http://example.com/codeflow?start=8&length=47)

Please check coverage result [https://dev.azure.com/mssonic/build/_build/results?buildId=513942&view=codecoverage-tab](https://dev.azure.com/mssonic/build/_build/results?buildId=513942&view=codecoverage-tab). Since these functions are intended to be reused as library functions, please cover it 100%.

File: utilities_common/cli.py
Comment (qiluo-msft) [inline utilities_common/cli.py:471]: >switchport_mode = portchannel['mode'] [](http://example.com/codeflow?start=8&length=37)

Could you add a testcase to cover this line?

File: utilities_common/cli.py
Comment (qiluo-msft) [inline utilities_common/cli.py:427]: >return [](http://example.com/codeflow?start=8&length=6)

Could you add a testcase to cover this line?

File: utilities_common/cli.py
Comment (sabakram) [reply utilities_common/cli.py:471]: Hi @qiluo-msft Qi, we have already added testcases to cover this point.

File: utilities_common/cli.py
Comment (qiluo-msft) [reply utilities_common/cli.py:427]: This is your new code :)
If you mean you copied code from elsewhere, it is still your responsibility to test/cover it.

Comment (sabakram) [reply utilities_common/cli.py:427]: Okay will add more test cases as well

**Review by qiluo-msft — APPROVED** [review]

**Comment by qiluo-msft** [comment]
Waiting on @prsunny approval.

File: config/main.py
Comment (prsunny) [inline config/main.py:4644]: Where is this checked now?

Comment (sabakram) [reply config/main.py:4644]: We have added a check for switchport mode routed, a routed port fulfill this behavior that a interface has a vlan membership or not. So this check becomes redundant after addition of new check.

**Review by prsunny — APPROVED** [review]

**Pull request merged**
